src/comm.cc

```diff
@@ -877,7 +879,7 @@ _comm_close(int fd, char const *file, int line)
     // For simplicity sake, we remain in the caller's context while still
     // allowing individual advanced callbacks to overwrite it.
 
-    if (F->ssl) {
+    if (F->ssl && !F->flags.harshClosureRequested) {
         const auto startCall = asyncCall(5, 4, "commStartTlsClose",
                                          callDialer(commStartTlsClose, fd));
```

Comment:
While what you write seems reasonable, it is not clear that RST is the only result of this function being called. It can and is also called for the FIN cases where TLS should be completed cleanly.

Reply:
> RST is the only result of this function being called

FWIW, the function (commStartTlsClose()) is not called for RST cases (F->flags.harshClosureRequested is false) but called for other (i.e., FIN) cases, as you noted.

Reply:
> Amos: it is not clear that RST is the only result of this function being called. It can and is also called for the FIN cases where TLS should be completed cleanly.

If "this function" is commStartTlsClose():

* In official code, before this PR: commStartTlsClose() was called in RST and FIN cases, and that was wrong.
* In PR code, commStartTlsClose() is only called in FIN cases, as intended.

RST is not the "result of calling this function".

If "this function" is commConfigureLinger():

* In official code, before this PR: commConfigureLinger() was called in both RST and FIN cases. That function call had no effect on commStartTlsClose() calls, and that was wrong.
* In PR code: commConfigureLinger() is called in both RST and FIN cases. That function call bans future commStartTlsClose() calls in RST cases only, as intended.

RST is the "result of calling this function" with `On` parameter, followed by a close(2) system call.

It is, of course, possible that some code paths that end with commStartTlsClose() are missing a commConfigureLinger() call to RST the connection, but that possible problem is outside _this_ PR scope. This PR does not change and should not change when/where commConfigureLinger() is called.